### ISSUE TITLE
Can't assign to ()

### ISSUE THREAD

**Issue by tyehle** [python/mypy#5594]
Mypy incorrectly complains about assignments to `()`. The simplest example of this is
```python
() = []
```
This code executes without a problem, but mypy says `error: can't assign to ()`

A more real world use case might be code like this
```python
(a, b), () = [[1, 2], []]
```
In this case the assignment to `()` is a concise way of asserting that the second list is empty.

* What is the actual behavior/output?
`error: can't assign to ()`
* What is the behavior/output you expect?
No type checking errors
* What are the versions of mypy and Python you are using?
A build from the latest commit as of writing this, 614090b7
* What are the mypy flags you are using?
No flags

If this is a change you decide would be good then I believe it could be implemented by removing the  the check on line 1930 of `mypy/semanal.py` and updating the tests.

**Comment by JelleZijlstra** [comment]
Thanks for the report! This seems like a real bug, but low-priority because little real-world code would do it. If somebody submits a PR to fix it, we'll take a look.

**Comment by PeterJCLaw** [comment]
Note that assignment to an empty tuple isn't valid syntax Python in 3.5, so `mypy` isn't totally wrong. It does seem to be valid syntax in 3.6 though.

Not sure if I should report this separately or just add it here -- assignment to `[]` is disallowed by `mypy`, but _is_ valid in 3.5 (and 3.6, I don't have other versions to hand to test with).
Assuming that that spelling is valid in all supported versions it would be great if `mypy` could allow it. Currently for code `[] = foo()` mypy emits the error `can't assign to ()`, which is doubly confusing as that source code doesn't feature any tuples.

For reference, I hit this because I wanted to unpack a (nested) iterable I expected to be empty:
``` python
[], kwargs = mock_thing.call_args
```
